sigproxy/main.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"syscall"
+)
+
+func checkArguments() {
+	if len(os.Args) == 1 {
+		log.Fatal("USAGE: sigproxy <command> [arguments ...]")
+	}
+}
+
+func installSignalHandler() chan os.Signal {
+	sc := make(chan os.Signal, 1)
+	signal.Notify(sc, syscall.SIGINT)
+
+	return sc
+}
+
+func execCommand() *exec.Cmd {
+	// First argument is command to run, remainder are its arguments
+	cmd := exec.Command(os.Args[1], os.Args[2:]...)
+
+	// These conveniently default to /dev/null otherwise...
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	if err := cmd.Start(); err != nil {
+		log.Fatal(err)
+	}
+
+	return cmd
+}
+
+func forwardSignals(sc chan os.Signal) {
+	for {
+		// Signalling PID 0 delivers to our process group
+		syscall.Kill(0, (<-sc).(syscall.Signal))
+	}
+}
+
+func waitAndExit(cmd *exec.Cmd) {
+	if err := cmd.Wait(); err != nil {
+		// Exit status is platform specific so not readily accessible - casts
+		// required to access system-dependent exit information
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			if waitStatus, ok := exitErr.Sys().(syscall.WaitStatus); ok {
+				os.Exit(waitStatus.ExitStatus())
```

Comment:
Good question. I think on Linux it's safe to do an unchecked cast for both `syscall.Signal` and `exitErr.Sys()`; the `err` from `cmd.Wait()` can be something other than an `ExitError` though. Do we have a general policy on this kind of thing?